sql/core/src/test/resources/sql-tests/inputs/pgSQL/boolean.sql

```diff
@@ -0,0 +1,265 @@
+
+--
+-- sanity check - if this fails go insane!
+--
+SELECT 1 AS one;
+
+
+-- ******************testing built-in type bool********************
+
+-- check bool input syntax
+
+SELECT true AS true;
+
+SELECT false AS false;
+
+SELECT cast('t' as boolean) AS true;
+
+SELECT cast('   f           ' as boolean) AS false;
+
+SELECT cast('true' as boolean) AS true;
+
+SELECT cast('test' as boolean) AS error;
+
+SELECT cast('false' as boolean) AS false;
+
+SELECT cast('foo' as boolean) AS error;
+
+SELECT cast('y' as boolean) AS true;
+
+SELECT cast('yes' as boolean) AS true;
+
+SELECT cast('yeah' as boolean) AS error;
+
+SELECT cast('n' as boolean) AS false;
+
+SELECT cast('no' as boolean) AS false;
+
+SELECT cast('nay' as boolean) AS error;
+
+SELECT cast('on' as boolean) AS true;
+
+SELECT cast('off' as boolean) AS false;
+
+SELECT cast('of' as boolean) AS false;
+
+SELECT cast('o' as boolean) AS error;
+
+SELECT cast('on_' as boolean) AS error;
+
+SELECT cast('off_' as boolean) AS error;
+
+SELECT cast('1' as boolean) AS true;
+
+SELECT cast('11' as boolean) AS error;
+
+SELECT cast('0' as boolean) AS false;
+
+SELECT cast('000' as boolean) AS error;
+
+SELECT cast('' as boolean) AS error;
+
+-- and, or, not in qualifications
+
+SELECT cast('t' as boolean) or cast('f' as boolean) AS true;
+
+SELECT cast('t' as boolean) and cast('f' as boolean) AS false;
+
+SELECT not cast('f' as boolean) AS true;
+
+SELECT cast('t' as boolean) = cast('f' as boolean) AS false;
+
+SELECT cast('t' as boolean) <> cast('f' as boolean) AS true;
+
+SELECT cast('t' as boolean) > cast('f' as boolean) AS true;
+
+SELECT cast('t' as boolean) >= cast('f' as boolean) AS true;
+
+SELECT cast('f' as boolean) < cast('t' as boolean) AS true;
+
+SELECT cast('f' as boolean) <= cast('t' as boolean) AS true;
+
+-- explicit casts to/from text
+SELECT cast(cast('TrUe' as string) as boolean) AS true, cast(cast('fAlse' as string) as boolean) AS false;
+SELECT cast(cast('    true   ' as string) as boolean) AS true,
+       cast(cast('     FALSE' as string) as boolean) AS false;
+SELECT cast(cast(true as boolean) as string) AS true, cast(cast(false as boolean) as string) AS false;
+
+SELECT cast(cast('  tru e ' as string) as boolean) AS invalid;    -- error
+SELECT cast(cast('' as string) as boolean) AS invalid;            -- error
+
+CREATE TABLE BOOLTBL1 (f1 boolean) USING parquet;
+
+INSERT INTO BOOLTBL1 VALUES (cast('t' as boolean));
+
+INSERT INTO BOOLTBL1 VALUES (cast('True' as boolean));
+
+INSERT INTO BOOLTBL1 VALUES (cast('true' as boolean));
+
+
+-- BOOLTBL1 should be full of true's at this point
+SELECT '' AS t_3, BOOLTBL1.* FROM BOOLTBL1;
+
+
+SELECT '' AS t_3, BOOLTBL1.*
+   FROM BOOLTBL1
+   WHERE f1 = cast('true' as boolean);
+
+
+SELECT '' AS t_3, BOOLTBL1.*
+   FROM BOOLTBL1
+   WHERE f1 <> cast('false' as boolean);
+
+-- SELECT '' AS zero, BOOLTBL1.*
+--    FROM BOOLTBL1
+--    WHERE booleq(cast('false' as boolean), f1);
```

Comment:
We have workaround here: `booleq(cast('false' as boolean), f1)` => `cast('false' as boolean) = f1`, or add new built-in udf `booleq`?

Reply:
Is `booleq` PostgreSQL specific?

Reply:
Yes. It's PostgreSQL specific. I have checked Presto, Redshift and Vertica.
MySQL, Oracle, SQL Server don't have boolean type.

Reply:
Instead of creating built-in UDF functions, can we create these UDFs in our SQLQueryTestSuite?

Reply:
Good suggestion